lib/Common/DataStructures/DictionaryEntry.h

```diff
@@ -118,11 +118,23 @@ namespace JsUtil
         TKey const& Key() const  { return key; }
     };
 
+    template <class TKey>
+    class BaseKeyEntry : public ValueEntry<TKey>
```

Comment:
What is the purpose of this class? A key value entry where the key is the same as the value?

Reply:
I'm just using this as a "`TKey` valued thing and a next index pair", which is the same as a "`TValue` valued thing and a next index pair" which is what `ValueEntry` is.